tools/experiment.py

```diff
@@ -696,26 +699,37 @@ def run_command_list(self, dryrun: bool = False):
             else:
                 print(" ".join(command))
 
-            self._previous_cmd_entry = command[1]
+            self._previous_cmd_entry.append(command[1])
 
         if not dryrun:
             organize_exp_result(self._workspace, self._command_var)
 
     def _prepare_run_command(self, command: List[str]) -> bool:
         self.set_arguments_to_cmd(command, "--workspace", str(self._workspace))
         cmd_entry = command[1]
+        previous_cmd = None
+        for previous_cmd in reversed(self._previous_cmd_entry):
+            if previous_cmd != "eval":
+                break
+
         if cmd_entry == "train":
             self.set_arguments_to_cmd(command, "--seed", str(self._repeat_idx))
         elif cmd_entry == "eval":
-            if self._previous_cmd_entry in self.OUTPUT_FILE_NAME:
-                file_path = self._find_model_path(self._previous_cmd_entry)
+            if previous_cmd in ["export", "optimize"]:
+                file_path = self._find_model_path(previous_cmd)
                 if file_path is None:
                     return False
                 self.set_arguments_to_cmd(command, "--load-weights", str(file_path))
                 output_path = str(file_path.parents[1])
             else:
                 output_path = str(self._workspace / "outputs" / "latest_trained_model")
             self.set_arguments_to_cmd(command, "--output", output_path)
+        elif cmd_entry == "optimize":
+            if previous_cmd == "export":
+                file_path = self._find_model_path(previous_cmd)
+                if file_path is None:
+                    return False
+                self.set_arguments_to_cmd(command, "--load-weights", str(file_path))
```

Comment:
Looks like a new feature to help PTQ `optimize` just after the `export`. Looks nice, but what if it's like `train` -> `eval` -> `optimize`? (I suppose QAT is intended?)

Reply:
Then this if statement won't executed and QAT will be executed using trained model weight (weights.pth).

Reply:
I see, thanks!

Reply:
How about adding single comment line ? 
```python3
if previous_cmd == "export": # PTQ will executed, else QAT will be executed
    file_path = self._find_model_path(previous_cmd)
```

Reply:
Done at https://github.com/openvinotoolkit/training_extensions/pull/2696/commits/f231b1e995d6bcb45570d76ced66bf15fd7f7ec2